src/util/misc/principal-type.ts

```diff
@@ -0,0 +1,11 @@
+export enum PrincipalType {
+  USER = "user",
+  APP = "app",
+}
+
+export const allPrincipalTypes = [PrincipalType.USER, PrincipalType.APP];
+
+export const principalMessaging = {
+  [PrincipalType.USER]: "user type",
+  [PrincipalType.APP]: "app type",
```

Comment:
Do we really need this? It seems that simply sticking to PrincipalType value is enough.

Reply:
Agreed, I was going to say the same!